snakemake/workflow.py

```diff
@@ -1196,11 +1196,26 @@ def decorate(ruleinfo):
                 rule.benchmark = ruleinfo.benchmark
             if not self.run_local and ruleinfo.group is not None:
                 rule.group = ruleinfo.group
+
+            # Using a snakemake-wrapper
             if ruleinfo.wrapper:
-                rule.conda_env = snakemake.wrapper.get_conda_env(
-                    ruleinfo.wrapper, prefix=self.wrapper_prefix
-                )
-                # TODO retrieve suitable singularity image
+
+                # Instantiate the wrapper to interact with scripts, etc.
+                rule_wrapper = WrapperSpec(ruleinfo.wrapper, prefix=self.wrapper_prefix)
```

Comment:
The update here is that we instantiate a WrapperSpec before interacting with it.